csharp-advanced-topics/UsingLazyInCSharp/HowtoUseLazyInCSharp/Cache.cs

```diff
@@ -0,0 +1,28 @@
+﻿
+namespace HowtoUseLazyInCSharp
+{
+    public class Cache
+    {
+        private Dictionary<string, object> cacheDictionary;
+
+        public Cache()
+        {
+            cacheDictionary = new Dictionary<string, object>();
+        }
+
+        public void Add(string key, object value)
+        {
+            cacheDictionary[key] = value;
+        }
+
+        public object Get(string key)
+        {
+            if (cacheDictionary.ContainsKey(key))
```

Comment:
This should use Dictionary.TryGetValue rather than performing a double key lookup in the expected/happy path.